src/functions/asyncCredential/asyncCredentialHandler.ts

```diff
@@ -111,7 +112,37 @@ export async function lambdaHandler(
   }
 
   const clientCredentials =
-    clientCredentialResponse.value as IClientCredentials;
+    clientCredentialResponse.value as IRegisteredClientCredentials;
+
+  const validateClientCredentialsResponse = clientCredentialsService.validate(
+    clientCredentials,
+    JSON.parse(requestBody),
+  );
+  if (validateClientCredentialsResponse.value != null) {
```

Comment:
We should use the isError property on this, rather than the value property please

Reply:
Typescript still wants a null check on the value but I can add in the error property as well

Reply:
We could cast this if we're confident in the value (which we should be as we control the types). It's how we've historically done this to retrieve the value.

Reply:
Updated to use `isError` property.